diff --git a/src/xml/LibXMLParser.cpp b/src/xml/LibXMLParser.cpp
index d832c640f..f4c4a78f1 100644
--- a/src/xml/LibXMLParser.cpp
+++ b/src/xml/LibXMLParser.cpp
@@ -198,7 +198,7 @@ LibXMLParser::reportError (const XMLErrorCode_t code,
     // We have no error log, but we shouldn't gloss over this error.  Use
     // the measure of last resort.
 
-    xmlErrorPtr libxmlError = xmlGetLastError();
+    auto libxmlError = xmlGetLastError();
 
     cerr << XMLError::getStandardMessage(code) << " at line and column numbers ";
 
@@ -396,7 +396,7 @@ LibXMLParser::parseNext ()
 
   if ( xmlParseChunk(mParser, mBuffer, bytes, done) )
   {
-    xmlErrorPtr libxmlError = xmlGetLastError();
+    auto libxmlError = xmlGetLastError();
 
     // I tried reporting the message from libXML that's available in
     // libxmlError->message, but the thing is bogus: it will say things
